File: sdks/python/api/extension/res.py
from typing import Optional, List, Dict, Union

from api.base_model import CyanBaseModel

from api.core.cyan_res import CyanRes
from api.core.question_res import QuestionRes


class ExtensionValidRes(CyanBaseModel):
    valid: Optional[str]


class ExtensionFinalRes(CyanBaseModel):
    cyan: CyanRes
    type: str = "final"


class ExtensionQnARes(CyanBaseModel):
    deterministic_state: List[Dict[str, str]]
    question: QuestionRes
    type: str = "questionnaire"


ExtensionRes = Union[ExtensionQnARes, ExtensionFinalRes]

File: sdks/node/src/api/extension/res.ts
import type { CyanRes } from '../core/cyan_res.js';
import type { QuestionRes } from '../core/question_res.js';

interface ExtensionValidRes {
  valid: string | null;
}

interface ExtensionFinalRes {
  cyan: CyanRes;
  type: 'final'; // Type discriminator value
}

interface ExtensionQnARes {
  deterministicState: Record<string, string>[];
  question: QuestionRes;
  type: 'questionnaire'; // Type discriminator value
}

type ExtensionRes = ExtensionQnARes | ExtensionFinalRes;

// export all
export type { ExtensionRes, ExtensionQnARes, ExtensionFinalRes, ExtensionValidRes };
